pkg/operators/formatters/formatters.go

```diff
@@ -180,6 +184,79 @@ func formatDuration(d time.Duration) string {
 	}
 }
 
+// Standard Linux file open flags from <fcntl.h>
+const (
+	// Access modes (handled separately)
+	O_RDONLY  = 0
+	O_WRONLY  = 1
+	O_RDWR    = 2
+	O_ACCMODE = 3
+
+	// Bit flags
+	O_CREAT     = 0o100
+	O_EXCL      = 0o200
+	O_NOCTTY    = 0o400
+	O_TRUNC     = 0o1000
+	O_APPEND    = 0o2000
+	O_NONBLOCK  = 0o4000
+	O_DSYNC     = 0o10000
+	O_FASYNC    = 0o20000
+	O_DIRECT    = 0o40000
+	O_LARGEFILE = 0o100000
+	O_DIRECTORY = 0o200000
+	O_NOFOLLOW  = 0o400000
+	O_NOATIME   = 0o1000000
+	O_CLOEXEC   = 0o2000000
+)
+
+// flagMap pairs the bitmask of a flag with its string representation.
+// Using a slice of structs makes the relationship explicit and order-independent.
```

Comment:
Is there a (large) performance hit when using a `map[int32]string` here instead?

Reply:
I don't have numbers on this specific part, but the overall change is "~2.6x speedup, uses 43% less memory, and requires only one memory allocation per operation compared to three previously." from #4806